### PR TITLE
[MU3 Backend] MusicXML compiler warnings

### PR DESCRIPTION
**Pull request by iveshenry18**
Resolves: Compiler warnings mentioned in #8492, #8325, and #8284

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: importexport/musicxml/importmxmlpass2.cpp [hunk 1]
```diff
@@ -4925,7 +4926,7 @@ Note* MusicXMLParserPass2::note(const QString& partId,
             else if (_e.name() == "stem")
                   stem(stemDir, noStem);
             else if (_e.name() == "type") {
-                  small = _e.attributes().value("size") == "cue" | _e.attributes().value("size") == "grace-cue";
+                  small = _e.attributes().value("size") == "cue" || _e.attributes().value("size") == "grace-cue";
```
Comment (Jojo-Schmitz) [inline importexport/musicxml/importmxmlpass2.cpp:4929]: Better also add parantheses to the expression?

Comment (iveshenry18) [reply importexport/musicxml/importmxmlpass2.cpp:4929]: It's not necessary (equality operators have precedence over logical OR, and bitwise OR for that matter. But if it'll help everyone sleep at night I'm happy to!

Comment (Jojo-Schmitz) [reply importexport/musicxml/importmxmlpass2.cpp:4929]: I added them first to stop the compiler warning, then detected the | vs || bug and fixed that too.

Comment (Jojo-Schmitz) [reply importexport/musicxml/importmxmlpass2.cpp:4929]: I checked now thogh: it does. Still those parens won't harm. OTH it shows that compiler warnings sometimes even point at a bug ;-)

File: importexport/musicxml/importmxmlpass2.cpp [hunk 2]
```diff
@@ -4926,7 +4926,7 @@ Note* MusicXMLParserPass2::note(const QString& partId,
             else if (_e.name() == "stem")
                   stem(stemDir, noStem);
             else if (_e.name() == "type") {
-                  small = _e.attributes().value("size") == "cue" | _e.attributes().value("size") == "grace-cue";
+                  small = (_e.attributes().value("size") == "cue") || (_e.attributes().value("size") == "grace-cue");
```
Comment (Jojo-Schmitz) [inline importexport/musicxml/importmxmlpass2.cpp:4929]: Hmm, I was suggesting the parenthezise the entire || expression, not their elements

**Pull request merged**